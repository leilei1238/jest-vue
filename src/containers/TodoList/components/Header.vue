<template>
  <div class="header">
    <div class="header-content">
      TodoList
      <input
        class="header-input"
        type="text"
        data-test="input"
        v-model="inputValue"
        placeholder="addTodoItem"
        @keyup.enter="addTodoItem"
      />
    </div>
  </div>
</template>

<script>
export default {
  name: 'Header',
  data: () => {
    return {
      inputValue: ''
    }
  },
  methods: {
    addTodoItem() {
      if (this.inputValue) {
        this.$emit('add', this.inputValue)
        this.inputValue = ''
      }
    }
  }
}
</script>

<style scoped lang="stylus">
.header{
  line-height 60px
  background #666
}
.header-content{
  width 600px
  margin 0 auto
  color #fff
  font-size 24px
}
.header-input{
  float right
  width 360px
  line-height 24px
  margin-top 16px
  outline none
  color #333333
}
</style>
