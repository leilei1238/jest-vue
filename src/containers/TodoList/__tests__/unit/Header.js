import { shallowMount } from '@vue/test-utils'
import Header from '../../components/Header.vue'

describe('Header 包含input框', () => {
  it('组件渲染正常', () => {
    const wrapper = shallowMount(Header)
    const input = wrapper.find('[data-test="input"]')
    expect(input.exists()).toBe(true)
  })
  it('Header中input框初始内容为空', () => {
    const wrapper = shallowMount(Header)
    const inputValue = wrapper.vm.$data.inputValue
    expect(inputValue).toBe('')
  })
  it('Header中input发生变化，数据应该跟着变', () => {
    const wrapper = shallowMount(Header)
    const input = wrapper.find('[data-test="input"]')
    input.setValue('hello')
    const inputValue = wrapper.vm.$data.inputValue
    expect(inputValue).toBe('hello')
  })

  it('Header中input框输入回车，无内容时，无反应', () => {
    const wrapper = shallowMount(Header)
    const input = wrapper.find('[data-test="input"]')
    input.setValue('')
    input.trigger('keyup.enter')
    expect(wrapper.emitted().add).toBeFalsy()
  })
  it('Header中input框输入回车，有内容时，向外触发事件', () => {
    const wrapper = shallowMount(Header)
    const input = wrapper.find('[data-test="input"]')
    input.setValue('test')
    input.trigger('keyup.enter')
    expect(wrapper.emitted().add).toBeTruthy()
  })
  it('Header中input框输入回车，有内容时，向外触发事件，同时清空inputValue', () => {
    const wrapper = shallowMount(Header)
    const input = wrapper.find('[data-test="input"]')
    input.setValue('test')
    input.trigger('keyup.enter')
    expect(wrapper.vm.$data.inputValue).toBe('')
  })
})
