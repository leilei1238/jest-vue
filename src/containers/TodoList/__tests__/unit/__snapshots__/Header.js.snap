// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 样式发生改变，做提示 1`] = `
<div class="header">
  <div class="header-content">
    TodoList
    <input type="text" data-test="input" placeholder="addTodoItem" class="header-input"></div>
</div>
`;
